pkg/provisioner/templates/kubernetes.go

```diff
@@ -147,12 +153,116 @@ echo "you can now access the cluster with:"
 echo "ssh -i <your-private-key> ubuntu@${INSTANCE_ENDPOINT_HOST}"
 `
 
+const NVKindTemplate = `
+
+: ${KIND_K8S_TAG:={{.Version}}}
+
+# The kind image to use. This image will be built if it is not available.
+: ${KIND_IMAGE:="kindest/node:${KIND_K8S_TAG}"}
+
+# The kubernetes tag to build the kind cluster from
+# From https://github.com/kubernetes/kubernetes/tags
+# See also https://hub.docker.com/r/kindest/node/tags
+: ${KIND_K8S_REPO:="https://github.com/kubernetes/kubernetes.git"}
+
+ARCH=$(uname -m)
+if [ "$ARCH" == "x86_64" ]; then
+  ARCH="amd64"
+fi
+if [ "$ARCH" == "aarch64" ]; then
+  ARCH="arm64"
+fi
+
+# Go
+# Set GO version
+GO_VERSION="1.23.6"
+wget https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz -O /tmp/go${GO_VERSION}.linux-amd64.tar.gz
+sudo rm -rf /usr/local/go
+sudo tar -C /usr/local -xzf /tmp/go${GO_VERSION}.linux-amd64.tar.gz
+
+# Add Go to PATH
+if ! grep -q 'export PATH="/usr/local/go/bin:$PATH"' ~/.bashrc; then
+  echo 'export PATH="/usr/local/go/bin:/$HOME/go/bin:$PATH"' >> ~/.bashrc
+fi
+export PATH="/usr/local/go/bin:$HOME/go/bin:$PATH"
+
+# Make
+install_packages_with_retry build-essential
+
+# Install kubectl
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+chmod +x kubectl
+sudo mv kubectl /usr/local/bin/
+
+# Install kind
+curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.26.0/kind-linux-${ARCH}
+chmod +x ./kind
+sudo install ./kind /usr/local/bin/kind
+
+# Install docker buildx, needed for building the kind image
+install_packages_with_retry docker-buildx-plugin
```

Comment:
Retries like this can save so much time/noise down the line. Nice.

Where do the retry helpers used in this PR come from?

Reply:
https://github.com/NVIDIA/holodeck/blob/main/pkg/provisioner/templates/common.go